tests/generated_data/generate_md.py

```diff
@@ -0,0 +1,137 @@
+    "5822582e7072996c1eef1cec24b61115d364987faa486659fe3d3dce8dae2aba",
+    "09d440e3725cec247dcb8703b646a87dd2a4d75343e8095c036c32795eefe3b9",
+    "3458204ed467519c19a5316eb278b5608472a1bbf15850ebfb462d5315e4f86d",
+    "2be5c21e3614f9f178fb49c4a34d0c18ffac30abd14ced917c60a52c8d8094b7",
+]
+
+keys: Dict[str, Key] = {}
+for i in range(4):
+    keys[f"ed25519_{i}"] = Key.from_securesystemslib_key(
+        {
+            "keytype": "ed25519",
+            "scheme": "ed25519",
+            "keyid": keyids[i],
+            "keyval": {
+                "public": public_values[i],
+                "private": private_values[i],
+            },
+        }
+    )
+
+expires_str = "2050-01-01T00:00:00Z"
+EXPIRY = datetime.strptime(expires_str, "%Y-%m-%dT%H:%M:%SZ")
+SPEC_VERSION = ".".join(SPECIFICATION_VERSION)
+OUT_DIR = "generated_data/ed25519_metadata"
+if not os.path.exists(OUT_DIR):
+    os.mkdir(OUT_DIR)
+
+SERIALIZER = JSONSerializer()
+key_info = copy.deepcopy(keys["ed25519_0"].to_securesystemslib_key())
+key_info["keyval"]["private"] = private_values[0]
+SIGNER = SSlibSigner(key_info)
+
+
+def verify_generation(md: Metadata, path: str) -> None:
+    """Verify that newly generated file equals the locally stored one.
+
+    Args:
+        md: Newly generated metadata object.
+        path: Path to the locally stored metadata file.
+    """
+    with open(path, "rb") as f:
+        static_md_bytes = f.read()
+        md_bytes = md.to_bytes(SERIALIZER)
+        if static_md_bytes != md_bytes:
+            raise ValueError(
+                f"Generated data != local data at {path}. Generate a new "
+                + "metadata with 'python generated_data/generate_md.py'"
+            )
+
+
+def generate_all_files(
+    dump: Optional[bool] = False, verify: Optional[bool] = False
+) -> None:
+    """Generate a new repository and optionally verify it.
+
+    Args:
+        dump: Wheter to dump the newly generated files.
+        verify: Whether to verify the newly generated files with the
+            local staored.
+    """
+    roles = {role_name: Role([], 1) for role_name in TOP_LEVEL_ROLE_NAMES}
+    root = Root(1, SPEC_VERSION, EXPIRY, {}, roles, True)
+    root.add_key("root", keys["ed25519_0"])
+    root.add_key("timestamp", keys["ed25519_1"])
+    root.add_key("snapshot", keys["ed25519_2"])
+    root.add_key("targets", keys["ed25519_3"])
+
+    md_root: Metadata[Root] = Metadata(root, {})
+
+    timestamp = Timestamp(1, SPEC_VERSION, EXPIRY, MetaFile(1))
+    md_timestamp: Metadata[Timestamp] = Metadata(timestamp, {})
+
+    meta: Dict[str, MetaFile] = {"targets.json": MetaFile(1)}
+    snapshot = Snapshot(1, SPEC_VERSION, EXPIRY, meta)
+    md_snapshot: Metadata[Snapshot] = Metadata(snapshot, {})
+
+    targets = Targets(1, SPEC_VERSION, EXPIRY, {})
+    md_targets: Metadata[Targets] = Metadata(targets, {})
+
+    for md in [md_root, md_timestamp, md_snapshot, md_targets]:
+        assert isinstance(md, Metadata)
+        md.sign(SIGNER)
```

Comment:
Why do you sign all metadata files with the root key? Why not sign timestamp with timestamp, snapshot with snapshot and targets with targets keys?

Reply:
Changed that.
Initially it was to reuse the signer.

Reply:
I am not sure I understand. So initially you wanted to sign with one key on purpose?

Reply:
Yea, because the files are not supposed to be read as a stand-alone repository, but as separate independent files, and as such we can reuse the same key as many times as we want.
The only reason I defined 4 keys is because of the root roles.